usr/lib/soft_stdll/soft_specific.c

```diff
@@ -3051,3 +3075,540 @@ CK_RV token_specific_aes_cmac(STDLL_TokData_t *tokdata, CK_BYTE *message,
 
     return rv;
 }
+
+#ifndef NO_EC
+
+static CK_RV make_ec_key_from_params(const CK_BYTE *params, CK_ULONG params_len,
+                                     EC_KEY **key)
+{
+    const unsigned char *oid;
+    ASN1_OBJECT *obj = NULL;
+    EC_KEY *ec_key = NULL;
+    int nid;
+    CK_RV rc = CKR_OK;
+
+    oid = params;
+    obj = d2i_ASN1_OBJECT(NULL, &oid, params_len);
+    if (obj == NULL) {
+        TRACE_ERROR("curve not supported by OpenSSL.\n");
+        rc = CKR_CURVE_NOT_SUPPORTED;
+        goto out;
+    }
+
+    nid = OBJ_obj2nid(obj);
+    if (nid == NID_undef) {
+        TRACE_ERROR("curve not supported by OpenSSL.\n");
+        rc = CKR_CURVE_NOT_SUPPORTED;
+        goto out;
+    }
+
+    ec_key = EC_KEY_new_by_curve_name(nid);
+    if (ec_key == NULL) {
+       TRACE_ERROR("curve not supported by OpenSSL.\n");
+       rc = CKR_CURVE_NOT_SUPPORTED;
+       goto out;
+    }
+
+out:
+    if (obj != NULL)
+        ASN1_OBJECT_free(obj);
+
+    if (rc != CKR_OK) {
+        if (ec_key != NULL)
+            EC_KEY_free(ec_key);
```

Comment:
EC_KEY_free checks internally if the argument is null.

Reply:
dito